vmm/src/api/http/http_endpoint.rs

```diff
@@ -226,6 +225,51 @@ vm_action_put_handler_body!(VmCoredump);
 
 vm_action_put_handler_body_with_fds!(VmAddNet, NetConfig, fds);
 
+impl PutHandler for VmRestore {
+    fn handle_request(
+        &'static self,
+        api_notifier: EventFd,
+        api_sender: Sender<ApiRequest>,
+        body: &Option<Body>,
+        mut files: Vec<File>,
+    ) -> std::result::Result<Option<Body>, HttpError> {
+        if let Some(body) = body {
+            let mut restore_cfg: RestoreConfig = serde_json::from_slice(body.raw())?;
+
+            let mut fds = Vec::new();
+            if !files.is_empty() {
+                fds = files.drain(..).map(|f| f.into_raw_fd()).collect();
+            }
+            if let Some(ref mut nets) = restore_cfg.net_fds {
+                warn!("Ignoring FDs sent via the HTTP request body");
+                let mut start_idx = 0;
+                for restored_net in nets.iter_mut() {
+                    // If there are no more FDs to assign, set fds field to None
+                    if start_idx >= fds.len() || restored_net.num_fds == 0 {
+                        restored_net.fds = None;
+                    } else {
+                        let end_idx = std::cmp::min(start_idx + restored_net.num_fds, fds.len());
+                        restored_net.fds = Some(fds[start_idx..end_idx].to_vec());
+                        start_idx = end_idx;
+                    }
+                }
+                if start_idx >= fds.len() && !fds.is_empty() {
+                    warn!("Ignoring FDs sent via SCM_RIGHTS");
+                }
+            } else if !fds.is_empty() {
```

Comment:
We need to ensure the sum of `restored.net>num_fds` is the same as `files.len()` (e.g. what's being sent via SCM_RIGHTS). Otherwise, this means the HTTP request from users is not valid and error should be returned to inform users.

Reply:
Right. This ensuring is indirectly done in `validate()` fn (but for each net dev). So, I thought of filling up the fds into fds field and let validate() throw the error. But yeah, having a check here is good as it eliminated other conditions while assigning the fds.